arch/arm/include/inttypes.h

```diff
@@ -29,91 +29,92 @@
  * Pre-processor Prototypes
  ****************************************************************************/
 
+#define int +2
+#define long +4
+
+#if defined(__INT32_TYPE__) && __INT32_TYPE__ == 2
+#  define INT32_MOD
+#else
+#  define INT32_MOD l
+#endif
+
+#undef int
+#undef long
+
 #define PRId8       "d"
 #define PRId16      "d"
-#define PRId32      "ld"
+#define PRId32      #INT32_MOD ## "d"
 #define PRId64      "lld"
-
 #define PRIdPTR     "d"
 
 #define PRIi8       "i"
 #define PRIi16      "i"
-#define PRIi32      "li"
+#define PRIi32      #INT32_MOD ## "i"
 #define PRIi64      "lli"
-
 #define PRIiPTR     "i"
 
 #define PRIo8       "o"
 #define PRIo16      "o"
-#define PRIo32      "lo"
+#define PRIo32      #INT32_MOD ## "o"
 #define PRIo64      "llo"
-
 #define PRIoPTR     "o"
 
 #define PRIu8       "u"
 #define PRIu16      "u"
-#define PRIu32      "lu"
+#define PRIu32      #INT32_MOD ## "u"
 #define PRIu64      "llu"
-
 #define PRIuPTR     "u"
 
 #define PRIx8       "x"
 #define PRIx16      "x"
-#define PRIx32      "lx"
+#define PRIx32      #INT32_MOD ## "x"
 #define PRIx64      "llx"
-
 #define PRIxPTR     "x"
 
 #define PRIX8       "X"
 #define PRIX16      "X"
-#define PRIX32      "lX"
+#define PRIX32      #INT32_MOD ## "X"
 #define PRIX64      "llX"
-
 #define PRIXPTR     "X"
 
 #define SCNd8       "hhd"
 #define SCNd16      "hd"
-#define SCNd32      "ld"
+#define SCNd32      #INT32_MOD ## "d"
 #define SCNd64      "lld"
-
 #define SCNdPTR     "d"
 
 #define SCNi8       "hhi"
 #define SCNi16      "hi"
-#define SCNi32      "li"
+#define SCNi32      #INT32_MOD ## "i"
 #define SCNi64      "lli"
-
 #define SCNiPTR     "i"
 
 #define SCNo8       "hho"
 #define SCNo16      "ho"
-#define SCNo32      "lo"
+#define SCNo32      #INT32_MOD ## "o"
 #define SCNo64      "llo"
-
 #define SCNoPTR     "o"
 
 #define SCNu8       "hhu"
 #define SCNu16      "hu"
-#define SCNu32      "lu"
+#define SCNu32      #INT32_MOD ## "u"
 #define SCNu64      "llu"
-
 #define SCNuPTR     "u"
 
 #define SCNx8       "hhx"
 #define SCNx16      "hx"
-#define SCNx32      "lx"
+#define SCNx32      #INT32_MOD ## "x"
 #define SCNx64      "llx"
-
 #define SCNxPTR     "x"
 
 #define INT8_C(x)   x
 #define INT16_C(x)  x
-#define INT32_C(x)  x ## l
+#define INT32_C(x)  x ## INT32_MOD
 #define INT64_C(x)  x ## ll
 
 #define UINT8_C(x)  x
 #define UINT16_C(x) x
-#define UINT32_C(x) x ## ul
+#define UINT32_C(x) x ## u ## INT32_MOD
```

Comment:
Things will not work this way. You need additional macro to expend INT32_MOD with ##

Reply:
So, I prefer to keep it in the simple form.